File: 2022/Day_01/main.py
def main():
    with open("input.txt", "r") as f:
        calories = [0]
        i = 0
        for line in f:
            if line == "\n":
                i += 1
                calories.append(0)
            else:
                calories[i] += int(line)
    
    result = []

    # Part 1
    result.append(max(calories))
    
    # Part 2
    calories.sort(reverse=True)
    sum = calories[0] + calories[1] + calories[2]
    result.append(sum)

    with open("output.txt", "w") as f:
        for i in result:
            f.write(f"{i}\n")


if __name__ == "__main__":
    main()

File: 2022/Day_01/main.go
package main

import (
    "bufio"
    "fmt"
    "log"
    "os"
    "io"
    "strconv"
    "sort"
)

func main() {
    file, err := os.Open("input.txt")
    if err != nil {
        log.Fatal(err)
    }
    defer file.Close()

    reader := bufio.NewReader(file)
    calories := []int{0}
    i := 0
    for {
        line, _, err := reader.ReadLine()
        if err == io.EOF {
            break
        }

        lineStr := string(line[:])
        if len(line) == 0 {
            i++
            calories = append(calories, 0)
        } else {
            calorie, _ := strconv.Atoi(lineStr)
            calories[i] += calorie
        }
    }

    sort.Ints(calories)

    // Part 1
    fmt.Printf("%v\n", calories[len(calories)-1])

    // Part 2
    sum := calories[len(calories)-1] + calories[len(calories)-2] + calories[len(calories)-3]
    fmt.Printf("%v\n", sum)
}